version: "3.8"

services:
  redis:
    profiles:
      - redis
      - hyperion
    container_name: redis
    image: redis:7.0.11-alpine
    restart: on-failure
    volumes:
      - redis:/data
    ports:
      - "6379:6379"
    networks:
      - hyperion

  rabbitmq:
    profiles:
      - rabbitmq
      - hyperion
    container_name: rabbitmq
    image: rabbitmq:3.11.15-management
    restart: on-failure
    environment:
      - RABBITMQ_DEFAULT_USER=${RABBITMQ_USER}
      - RABBITMQ_DEFAULT_PASS=${RABBITMQ_PASSWORD}
      - RABBITMQ_DEFAULT_VHOST=${RABBITMQ_VHOST}
    volumes:
      - rabbitmq:/var/lib/rabbitmq
    ports:
      - "5672:5672"
      - "15672:15672"
    networks:
      - hyperion

volumes:
  rabbitmq:
    driver: local
    driver_opts:
      o: bind
      type: none
      device: /home/alcor/d/rabbitmq
  redis:
    driver: local
    driver_opts:
      o: bind
      type: none
      device: /home/alcor/d/redis

networks:
  hyperion:
    driver: bridge
